### PR TITLE
Add roadmap to docs

### PR DESCRIPTION
**Pull request by yurishkuro**

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Roadmap
+
+The following is only a selection of some of the major features we plan to implement in the near future.
```
Comment (vprithvi) [inline docs/roadmap.md]: I'd add the time horizon right after `near future`

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+[jaeger-ui/milestone/1](https://github.com/uber/jaeger-ui/milestone/1)
+
+## Instrumentation Libraries in More Languages
+
+[jaeger/issues/366](https://github.com/uber/jaeger/issues/366)
+
+## Data Pipeline
+
+Post-collection data pipeline for trace aggregation and data mining based on Apache Flink.
```
Comment (vprithvi) [inline docs/roadmap.md]: Link to issue?

File: docs/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+## Drop-in Replacement for Zipkin
+
+Features for Jaeger backend to be a drop-in replacement for Zipkin backend.
+
+[jaeger/milestone/2](https://github.com/uber/jaeger/milestone/2)
+
+## Path-Based Dependency Diagrams
+
+Current service dependency diagram in Jaeger only shows service-to-service links.
```
Comment (vprithvi) [inline docs/roadmap.md]: s/Current/The
`Current` doesn't add any additional information.

File: docs/roadmap.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+Such diagrams are of limited usefulness because they do not account for the actual
+execution paths passing through the service, where requests to one endpoint may
+involve one set of downstream dependencies which are different from dependencies
+of another endpoint. We plan to open source the aggregation module that builds
+path-based dependency diagrams with the following features:
+
+  1. show all upstream and downstream dependencies of a selected service `postmaster`,
```
Comment (vprithvi) [inline docs/roadmap.md]: Capitalize first letter, add full stops consistently.

File: docs/roadmap.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+     in the graph that are not relevant to requests passing through both `cli_user2`
+     and `postmaster`.
+
+<img src="../images/path-dependency.svg">
+
+## Latency Histograms
+
+Jaeger traces contain a wealth of information about the system executed a given request.
+But can one find interesting traces? Latency histograms allow not only navigation to the
+interesting tracing, such as those representing the long tail, but also analysis of the
```
Comment (vprithvi) [inline docs/roadmap.md]: s/tracing/traces

File: docs/roadmap.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+request paths from upstream services. In the screenshot below we see how brush-selecting
```
Comment (vprithvi) [inline docs/roadmap.md]: brush-selecting adds no additional information over saying selecting

File: docs/roadmap.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+The target time horizon for most of these features is 6-12 months.
+
+## Adaptive Sampling
+
+The most common way of using Jaeger client libraries is with probabilistic sampling which makes a determination
+if a new trace should be sampled or not. Sampling is necessary to control the amount of tracing data reaching
+the storage backend. There are two issues with the current approach:
+
+  1. The individual microservices have little insight into what the appropriate sampling rate should be.
```
Comment (vprithvi) [inline docs/roadmap.md]: s/The individual/Individual

File: docs/roadmap.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+Such diagrams are of limited usefulness because they do not account for the actual
+execution paths passing through the service, where requests to one endpoint may
+involve one set of downstream dependencies which are different from dependencies
+of another endpoint. We plan to open source the aggregation module that builds
+path-based dependency diagrams with the following features:
+
+  1. show all upstream and downstream dependencies of a selected service `postmaster`,
+     not just the immediate neighbors
+  1. can be shown at the service level or at the endpoint level
+  1. interactive, for example specifying a fiter `cli_user2` grays out the paths
```
Comment (vprithvi) [inline docs/roadmap.md]: s/fiter/filter

File: docs/roadmap.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+runs analysis on all traces collected in the backend, inspects then for known completeness
+and quality problems, and provides summary reports to service owners with suggestions on
+improving the quality metrics and links to sample traces that exhibit the issues.
+
+## Dynamic Configuration
+
+We need a dynamic configuration solution ([jaeger/issues/355](https://github.com/uber/jaeger/issues/355))
+that comes in handy in various scenarios:
+
+  * blacklisting services
```
Comment (vprithvi) [inline docs/roadmap.md]: Capitalization

File: docs/roadmap.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+request paths from upstream services. In the screenshot below we see how brush-selecting
+a portion of the histogram reveals the breakdowns of the endpoints and upstream callers
+that are responsible for the long tail.
+
+<img src="../images/latency-histrogram.png">
+
+## Trace Quality Metrics
+
+When deploying a distributed tracing solution like Jaeger in large organizations,
+especially those whose architectures are composed of many different technologies,
```
Comment (vprithvi) [inline docs/roadmap.md]: s/especially those/\

File: docs/roadmap.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+there are always questions about how much of the architecture is actually integrated
```
Comment (vprithvi) [inline docs/roadmap.md]: s/actually/\

File: docs/roadmap.md (hunk 2 above)
Comment (yurishkuro) [reply docs/roadmap.md]: there is not an issue yet

**Pull request merged**